internal/controller/rbac/definition/roles.go

```diff
@@ -72,6 +73,7 @@ func RenderClusterRoles(d *v1.CompositeResourceDefinition) []rbacv1.ClusterRole
 				Resources: []string{
 					d.Spec.Names.Plural,
 					d.Spec.Names.Plural + suffixStatus,
+					d.Spec.Names.Plural + suffixFinalizers,
```

Comment:
I'm not following why this is needed - can you add a comment (to the code) explaining?

This system role is going to allow Crossplane (the 'core' pod, not the RBAC manager pod) `*` access to any composite resource (XR) defined by an XRD. Crossplane does create resources that block the owner deletion of these resources (e.g. when creating a composed resource owned by the XR) but I would have thought the `*` verb already implied it could update finalizers for the XR.

Reply:
(Ditto for the claims variant below.)